src/main/java/com/google/cloud/spanner/pgadapter/ConnectionHandler.java

```diff
@@ -100,18 +105,26 @@ public class ConnectionHandler extends Thread {
   private WellKnownClient wellKnownClient;
   private ExtendedQueryProtocolHandler extendedQueryProtocolHandler;
 
-  ConnectionHandler(ProxyServer server, Socket socket) {
+  ConnectionHandler(ProxyServer server, ByteChannel byteChannel) throws IOException {
     super("ConnectionHandler-" + CONNECTION_HANDLER_ID_GENERATOR.incrementAndGet());
     this.server = server;
-    this.socket = socket;
+    this.socketChannel = byteChannel;
+    if (byteChannel instanceof SocketChannel
```

Comment:
What are the cases when byteChannel is not of SocketChannel type?

Reply:
I've added a comment to explain. `SocketChannel` is only used by TCP connections. Unix domain sockets use other implementations.